clang/lib/Sema/SemaCoroutine.cpp

```diff
@@ -1176,6 +1176,10 @@ void Sema::CheckCompletedCoroutineBody(FunctionDecl *FD, Stmt *&Body) {
   for (AddrLabelExpr *ALE : Fn->AddrLabels)
     Diag(ALE->getBeginLoc(), diag::err_coro_invalid_addr_of_label);
 
+  // Coroutines always return a handle, so they can't be [[noreturn]].
+  if (Fn->isCoroutine())
+    Diag(FD->getLocation(), diag::warn_noreturn_coroutine) << FD;
```

Comment:
```suggestion
  // Coroutines always return a handle, so they can't be [[noreturn]].
  Diag(FD->getLocation(), diag::warn_noreturn_coroutine) << FD;
```
I should really hope we don’t get here if this *isn’t* a coroutine. 👀

Reply:
Actually, I’m a bit slow today, now that I look at it, this was probably meant to be `Fn->isNoReturn()` wasn’t it.

Reply:
Yea, just noticed that I wanted to check if it's noreturn 😅